infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SnapshotSender.java

```diff
@@ -133,15 +135,11 @@ public void transmit(UUID snapshotSyncEventId) {
                     cancel = true;
                     break;
                 } catch (ReplicationReaderException e) {
-                   if (e.getCause() instanceof TimeoutException &&
-                       fsm.getSession().getSubscriber().getModel() == LogReplication.ReplicationModel.ROUTING_QUEUES) {
-                       log.info("Snapshot sync timed out waiting for data.  Will request a new snapshot sync");
-                       snapshotSyncCancel(snapshotSyncEventId, LogReplicationError.UNKNOWN, true);
-                   } else {
-                       log.error("Replication Reader Exception thrown from an unkown path", e);
-                       snapshotSyncCancel(snapshotSyncEventId, LogReplicationError.UNKNOWN, false);
-                   }
-                    cancel = true;
+                    if (e.getCause() instanceof TimeoutException) {
```

Comment:
we should not remove the check for routing queues model.  If a timeout exception gets thrown for any reason from another model in future, this can be incorrect.